core/src/test/java/org/infinispan/test/AbstractInfinispanTest.java

```diff
@@ -72,6 +113,191 @@
+         jgroupsMethodsAllowedToBlock(builder);
+
+         questionableMethodsAllowedToBlock(builder);
+
+         allowTestsToBlock(builder);
+
+         builder.blockingMethodCallback(bm ->
+            writeJUnitReport(new AssertionError(String.format("Blocking call! %s on thread %s", bm, Thread.currentThread())), "Blocking"));
+      }).install();
+
+      Thread.setDefaultUncaughtExceptionHandler((thread, t) -> {
+         LogFactory.getLogger("Infinispan-TEST").fatal("Throwable was not caught in thread " + thread +
+               " - exception is: " + t);
+         // RxJava propagates via this and we don't want to worry about it
+         if (!(t instanceof UndeliverableException)) {
+            writeJUnitReport(t, "Uncaught");
+         }
+      });
+   }
+
+   private static void methodsToBeRemoved(BlockHound.Builder builder) {
+      // Total Order is to be removed!
+      builder.allowBlockingCallsInside(StateConsumerImpl.class.getName(), "awaitTotalOrderTransactions");
+
+      // The internal map only supports local mode - we need to replace with Caffeine
+      builder.allowBlockingCallsInside(RecoveryManagerImpl.class.getName(), "registerInDoubtTransaction");
+   }
+
+   private static void jgroupsMethodsAllowedToBlock(BlockHound.Builder builder) {
+      // UDP calls PlainDatagramSocketImpl#send which can block if OS buffer is full - but we just ignore
+      builder.allowBlockingCallsInside(UDP.class.getName(), "_send");
+      builder.allowBlockingCallsInside(FlowControl.class.getName(), "adjustCredit");
+      builder.allowBlockingCallsInside(Table.class.getName(), "add");
+      builder.allowBlockingCallsInside(Table.class.getName(), "get");
+
+      // THIS IS A VERY BAD HACK - because you can't instrument a static block of the class in jgroups Version class
+      // which loads a properties file
+      builder.allowBlockingCallsInside(Properties.class.getName(), "load");
+
+      // The NonBlockingCreditMap being non blocking is a bit questionable as they hold the lock while doing a lot of
+      // work - including acquiring other locks and possibly calling await
+      builder.allowBlockingCallsInside(NonBlockingCreditMap.class.getName(), "decrement");
+      builder.allowBlockingCallsInside(NonBlockingCreditMap.class.getName(), "replenish");
+   }
+
+   private static void questionableMethodsAllowedToBlock(BlockHound.Builder builder) {
+      // Component registry has a lock to protect its state - is short lived lock
+      builder.allowBlockingCallsInside(BasicComponentRegistryImpl.class.getName(), "prepareWrapperChange");
+
+      // This one should probably not be allowed - it is waiting for another component to start
+      // TODO: This might actually be a bug in the rewiring logic. (shows in StateTransferOverwritingValueTest)
+      builder.allowBlockingCallsInside(BasicComponentRegistryImpl.class.getName(), "awaitWrapperState");
+
+      // Believe these can technically block for TCP when flushing to socket
```

Comment:
Not just that, but there are plenty of other protocols using locking in the JGroups stack...
Running with `transfer-queue-bundler` would remove the direct socket writes from `sendCommand`, but the bundler can still block writing to the limited queue.

Reply:
Good to know, I think having a list of known protocols that can block would help figure out what to do possibly.